pkg/karmadactl/join.go

```diff
@@ -278,7 +282,7 @@ func generateSecretInMemberCluster(clusterKubeClient kubeclient.Interface, clust
 	return clusterSecret, nil
 }
 
-func generateClusterInControllerPlane(controlPlaneConfig, clusterConfig *rest.Config, clusterName string, secret corev1.Secret) (*clusterv1alpha1.Cluster, error) {
+func generateClusterInControllerPlane(controlPlaneConfig, clusterConfig *rest.Config, clusterName string, provider string, secret corev1.Secret) (*clusterv1alpha1.Cluster, error) {
```

Comment:
I concern the parameter list is a little bit too long, can we use the `CommandJoinOption`?

Reply:
I think so too,in my own branch,in addition to 'provider',i also added a lot of fields,using in CommandJoinOption struct.
![image](https://user-images.githubusercontent.com/31329928/143419540-54ab97fe-46b9-4d9d-b3f7-607c20eb826d.png)

Reply:
> 
Do I need to change this?

Reply:
Yes, we can use CommandJoinOption to pass parameter in this case.

Reply:
+1